readthedocs/projects/tasks/builds.py

```diff
@@ -798,34 +793,44 @@ def store_build_artifacts(
         types_to_copy = []
         types_to_delete = []
 
-        # HTML media
-        if html:
-            types_to_copy.append(('html', self.data.config.doctype))
-
-        # Search media (JSON)
-        if search:
-            types_to_copy.append(('json', 'sphinx_search'))
-
-        if localmedia:
-            types_to_copy.append(('htmlzip', 'sphinx_localmedia'))
-        else:
-            types_to_delete.append('htmlzip')
-
-        if pdf:
-            types_to_copy.append(('pdf', 'sphinx_pdf'))
-        else:
-            types_to_delete.append('pdf')
-
-        if epub:
-            types_to_copy.append(('epub', 'sphinx_epub'))
-        else:
-            types_to_delete.append('epub')
+        artifact_types = ("html", "json", "htmlzip", "pdf", "epub")
+        for artifact_type in artifact_types:
+            if os.path.exists(
+                self.data.project.artifact_path(
+                    version=self.data.version.slug,
+                    type_=artifact_type,
+                )
+            ):
+                types_to_copy.append(artifact_type)
+            # Never delete HTML nor JSON (search index)
+            elif artifact_type not in ("html", "json"):
```

Comment:
I think it's probably clearer to have a constant like `UNDELETABLE_TYPES` that we're checking here. This seems like something that might get missed in a future update, and we should have a bit more comment on why we never delete these.

Reply:
_Good_ point. I added constants for this.